File: backend/app/schemas/document_file.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class DocumentFileBase(BaseModel):
    document_id: int
    file_name: str
    file_key: str
    file_url: Optional[str] = None
    mime_type: Optional[str] = None
    file_size: Optional[int] = None

class DocumentFileCreate(DocumentFileBase):
    pass

class DocumentFileUpdate(BaseModel):
    file_name: Optional[str] = None
    file_key: Optional[str] = None
    file_url: Optional[str] = None
    mime_type: Optional[str] = None
    file_size: Optional[int] = None

class DocumentFileRead(DocumentFileBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/Requests/DocumentFile.ts
export interface SaveFileRequest {
  document_id: number;
  file_name: string;
  file_key: string;
  file_url?: string | null;
  mime_type?: string;
  file_size?: number | null;
}

export interface CreateDocumentFileRequest {
  document_id: number;
  file_name: string;
  file_key: string;
  file_url: string;
  mime_type: string;
  file_size: number;
}